tests/unit/test_charm.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Learn more about testing at: https://juju.is/docs/sdk/testing
+
+import unittest
+
+from ops import testing
+from ops.model import ActiveStatus, BlockedStatus
+from ops.testing import Harness
+
+from charm import Route53AcmeOperatorCharm
+
+testing.SIMULATE_CAN_CONNECT = True
+
+
+class TestCharm(unittest.TestCase):
+    def setUp(self):
+        self.harness = Harness(Route53AcmeOperatorCharm)
+        self.harness.set_leader(True)
+        self.harness.set_can_connect("lego", True)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+        self.r_id = self.harness.add_relation("certificates", "remote")
```

Comment:
I don't think we want to add relations in the setup. It should probably be explicit in the tests that want this relation. And we also probably want to validate behavior for cases where there are no relations.

Reply:
We don't want to abstract everything in the setup. I'd also remove the `set_leader` and `set_connect`. Those are assumptions that won't always be true when we deploy the charm.

Reply:
You are right they were unnecessary.
For validating the behaviour in case of no relation, I'd say that should be done on the lib's side, as this charm is unaware of relations, and it goes to active if its required configuration is provided and passes validation :)
I think the lib doesn't consider that now.

Reply:
If the behavior is already validated in the lib side that's fine. In all cases I don't think adding relations and setting `can_connect` to `True` should be in the setUp.

Reply:
Yes, I removed it :)
No, I think the lib actually doesn't validate it, I'm not sure it really needs to, but it's definitely worth investigating.